app/support/reservations/duration_change_validations.rb

```diff
@@ -0,0 +1,32 @@
+class Reservations::DurationChangeValidations
+  include ActiveModel::Validations
+
+  validate :start_time_not_changed, unless: Proc.new { |r| r.reservation.reserve_start_at_editable? }
+  validate :duration_not_shortened, if: Proc.new { |r| Time.zone.now >= r.reservation.reserve_start_at }
+
+  def initialize(reservation)
+    @reservation = reservation
+  end
+
+  def reservation
+    @reservation
+  end
+
+  def copy_errors!
```

Comment:
Do we have access to `after_validation` that we could hook this method into? I'd like to see this happen automatically after running the validations. Maybe overriding `valid?` with a call to `super` in there.